src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -414,27 +414,58 @@
                                         (false? (:json_unfolding existing-field))))]
         (remove should-not-unfold? json-fields)))))
 
+(defn- sample-json-row-honey-sql
+  "Return a hoenysql used to get row sample to describe json columns.
+
+  If the talbe has a PK, try to fetch both first and last rows (see ##25744).
+  Else fetch the first n rows only"
+  [driver table-identifier json-field-names pk-identifier]
+  (binding [hx/*honey-sql-version* (sql.qp/honey-sql-version driver)]
+    (let [pk-expr          (sql.qp/maybe-wrap-unaliased-expr pk-identifier)
+          table-expr       (sql.qp/maybe-wrap-unaliased-expr table-identifier)
+          json-field-names (mapv sql.qp/maybe-wrap-unaliased-expr json-field-names)]
+      (if pk-identifier
+        {:select json-field-names
+         :from   [table-expr]
+         ;; mysql doesn't support limit in where, so we're using inner join here
+         :join  [[{:union [{:nest {:select   [pk-expr]
+                                   :from     [table-expr]
+                                   :order-by [[pk-expr :asc]]
+                                   :limit    (/ metadata-queries/nested-field-sample-limit 2)}}
+                           {:nest {:select   [pk-expr]
+                                   :from     [table-expr]
+                                   :order-by [[pk-expr :desc]]
+                                   :limit    (/ metadata-queries/nested-field-sample-limit 2)}}]}
+                  :result]
+                 [:=
+                  (sql.qp/maybe-wrap-unaliased-expr (hx/identifier :field :result :id))
+                  (hx/identifier :field table-identifier :id)]]}
+        {:select json-field-names
+         :from   [table-expr]
+         :limit  metadata-queries/nested-field-sample-limit}))))
```

Comment:
I'm not sure it's worth the complication:
1. we can't call row_number over a json column because the DBMS does not support cooperation operation over json columns, which means we need to order by some column other than json columns
```
json=# select *, row_number() over(order by json_bit) from json;
ERROR:  could not identify an ordering operator for type json
LINE 1: select *, row_number() over(order by json_bit) from json;
                                             ^
HINT:  Use an explicit ordering operator or modify the query.
```
2. I think most of the time, a table will have pks

1 alone make me don't want to go there because the order is uncertain and it's different from what we want -- `first and last rows`.  it makes the query less certain and too complicated IMO.

Reply:
Yes, makes sense. I incorrectly assumed `row_number() over()` would return the rows in the order they were inserted, but that's obviously not the case